src/telegrafs/actions/thunks.ts

```diff
@@ -108,6 +108,61 @@ export const createTelegraf = (telegraf: Telegraf) => async (
   }
 }
 
+export const cloneTelegraf = (telegraf: Telegraf) => async (
+  dispatch: Dispatch<Action>
+) => {
+  try {
+    // Step 1: create a new telegraf
+    const response = await postTelegraf({data: telegraf})
+
+    if (response.status !== 201) {
+      throw new Error(response.data.message)
+    }
+
+    const clonedTelegraf = response.data
+
+    // Step 2: clone the labels and post them with the new telegraf's id
+    const pendingLabels = telegraf.labels.map(labelID =>
+      postTelegrafsLabel({
+        telegrafID: clonedTelegraf.id,
+        data: {labelID},
+      })
+    )
+
+    const mappedLabels = await Promise.all(pendingLabels)
+
+    if (
+      mappedLabels.length &&
+      mappedLabels.some(label => label.status !== 201)
+    ) {
+      throw new Error(
+        'An error occurred cloning the labels for this telegraf config'
+      )
```

Comment:
It might be helpful to include the name of the label that failed to be cloned. Although it may be obvious if there are only a couple of labels, if there a bunch of them, and only one failed, it would make things easier to know which one failed.

Reply:
Good idea, but we don't have the label's name or any other information about the label when the status is not a success. We get back only an error object.